src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java

```diff
@@ -53,7 +50,10 @@ public UnmodifiableHeaders(Headers headers) {
 
     public boolean containsValue(Object value) { return headers.containsValue(value); }
 
-    public List<String> get(Object key) { return map.get(key); }
+    public List<String> get(Object key) {
+        List<String> l = headers.get(key);
+        return l == null ? null : Collections.unmodifiableList(l);
+    }
```

Comment:
You could simply return `map.get(k)` here - since map is an unmodifiable map wrapping  `unmodHeaders`